lib/Config.js

```diff
@@ -71,11 +71,24 @@ class Config {
 
         if (checkedConfig.hasOwnProperty('level')) {
             LogLevel.throwIfInvalid(checkedConfig.level);
+        }
+        if (checkedConfig.hasOwnProperty('dump')) {
+            LogLevel.throwIfInvalid(checkedConfig.dump);
+        }
+        // for check log level vs. log dump level
+        const newLogLevel = checkedConfig.level || this.logLevel;
+        const newLogDumpLevel = checkedConfig.dump || this.dumpThreshold;
+        if (newLogDumpLevel &&
+            !LogLevel.shouldLog(newLogDumpLevel, newLogLevel)) {
+            throw new Error(
+                'Logging level should be at most logging dump level');
+        }
+
+        if (checkedConfig.hasOwnProperty('level')) {
```

Comment:
The previous check at line 72 is to check correct level value. This check here (at line 87) is to set this level value if it's given in the input config.